File: go_files/ex7.go
package main

import "fmt"

type Professor struct {
	nome     string
	escolas  []*Escola 
}

func (p *Professor) AdicionarEscola(escola *Escola) {
	p.escolas = append(p.escolas, escola)
}

type Escola struct {
	nome      string
	professores []*Professor 
}

func (e *Escola) AdicionarProfessor(professor *Professor) {
	e.professores = append(e.professores, professor)
	professor.AdicionarEscola(e)
}

func (p Professor) Informar() string {
	return fmt.Sprintf("Professor: %s", p.nome)
}

func (e Escola) Informar() string {
	return fmt.Sprintf("Escola: %s", e.nome)
}

func main() {

	professor1 := &Professor{nome: "Guilherme"}
	professor2 := &Professor{nome: "Eduardo"}

	escola1 := &Escola{nome: "Escola A"}
	escola2 := &Escola{nome: "Escola B"}

	escola1.AdicionarProfessor(professor1)
	escola1.AdicionarProfessor(professor2)
	escola2.AdicionarProfessor(professor1) 


	fmt.Println(escola1.Informar())
	for _, prof := range escola1.professores {
		fmt.Println(prof.Informar())
	}

	fmt.Println("\n" + escola2.Informar())
	for _, prof := range escola2.professores {
		fmt.Println(prof.Informar())
	}
}

File: pyton_files/ex7.py


class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.escolas = []  

    def adicionar_escola(self, escola):
        if escola not in self.escolas:
            self.escolas.append(escola)
            escola.adicionar_professor(self)  

    def exibir_escolas(self):
        return ", ".join([escola.nome for escola in self.escolas])

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []  

    def adicionar_professor(self, professor):
        if professor not in self.professores:
            self.professores.append(professor)

    def exibir_professores(self):
        return ", ".join([professor.nome for professor in self.professores])


def main():
    escola1 = Escola("Escola Primária")
    escola2 = Escola("Escola Secundária")

    professor1 = Professor("Carlos")
    professor2 = Professor("Ana")

    
    professor1.adicionar_escola(escola1)
    professor1.adicionar_escola(escola2)
    professor2.adicionar_escola(escola1)

    print(f"{professor1.nome} leciona em: {professor1.exibir_escolas()}")
    print(f"{professor2.nome} leciona em: {professor2.exibir_escolas()}")

   
    print(f"Professores da {escola1.nome}: {escola1.exibir_professores()}")
    print(f"Professores da {escola2.nome}: {escola2.exibir_professores()}")


if __name__ == "__main__":
    main()
